psl/parser-database/src/attributes/id.rs

```diff
@@ -97,10 +97,15 @@ pub(super) fn model(model_data: &mut ModelAttributes, model_id: crate::ModelId,
         })
         .collect();
 
-    if !fields_that_are_not_required.is_empty() {
+    if !fields_that_are_not_required.is_empty() && !model_data.is_ignored {
         ctx.push_error(DatamodelError::new_model_validation_error(
             &format!(
-                "The id definition refers to the optional fields {}. ID definitions must reference only required fields.",
+                "The id definition refers to the optional field{} {}. ID definitions must reference only required fields.",
+                if fields_that_are_not_required.len() == 1 {
+                    ""
+                } else {
+                    "s"
+                },
```

Comment:
nit: This pluralisation logic is duplicated below. Perhaps it's useful to extract it as a common utility helper.